### PR TITLE
[JENKINS-34973] RejectedAccessException recorded more reliably

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-34973](https://issues.jenkins-ci.org/browse/JENKINS-34973): `ScriptApproval.accessRejected` not being called reliably. Downstream of https://github.com/jenkinsci/script-security-plugin/pull/243.

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/workflow/cps/SandboxContinuableTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+public class SandboxContinuableTest {
+
+    @ClassRule public static BuildWatcher buildWatcher = new BuildWatcher();
+
+    @Rule public JenkinsRule r = new JenkinsRule();
+
+    @Issue("JENKINS-34973")
+    @Test public void scriptApproval() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition("catchError {Jenkins.instance}", true));
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/cps/SandboxContinuableTest.java]: Adapted from https://github.com/jenkinsci/workflow-basic-steps-plugin/pull/51.

**Comment by jglick** [comment]
Windows agents…

**Comment by jglick** [comment]
[INFRA-2075](https://issues.jenkins-ci.org/browse/INFRA-2075) for CI issues.

**Comment by abayer** [comment]
@jglick - Script Security 1.58 was just released - could you update to use that? Thanks.

**Comment by jglick** [comment]
@abayer done

**Pull request merged**